bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -291,6 +381,100 @@ where
 			[GlobalConsensus(network), AccountKey20 { network: None, key: token.into() }],
 		)
 	}
+
+	/// Constructs an XCM message destined for AssetHub that withdraws assets from the sovereign
+	/// account of the Gateway contract and either deposits those assets into a recipient account or
+	/// forwards the assets to another parachain.
+	fn convert_send_native_token(
+		message_id: H256,
+		chain_id: u64,
+		token_id: TokenId,
+		destination: Destination,
+		amount: u128,
+		asset_hub_fee: u128,
+	) -> Result<(Xcm<()>, Balance), ConvertMessageError> {
+		let network = Ethereum { chain_id };
+		let asset_hub_fee_asset: Asset = (Location::parent(), asset_hub_fee / 2).into();
```

Comment:
What is the magic number 2 here? Looks like its not on the original PR: https://github.com/Snowfork/polkadot-sdk/pull/128/files#diff-5a39b1d9ef60342361183a0aed74e5dae96dad6a973162b59c593705a31e3314R396

Reply:
More details in https://github.com/Snowfork/polkadot-sdk/pull/128#discussion_r1741002931

Reply:
The problem described in https://github.com/Snowfork/polkadot-sdk/pull/128#discussion_r1741002931 makes sense. Seems weird that DepositReserveAsset and InitiateReserveWithdraw have different fee handling behavior.

Though I think I share Clara's unease over the magic number. Could do you more analysis to decide how the fee should be split between xcm execution and transport fees.

And the magic number should be made into a constant, and there should be comments explaining why we are doing this.

Reply:
>  analysis to decide how the fee should be split between xcm execution and transport fees

Let's take https://app.snowbridge.network/history#0x75765a2f18b00e83b79659c6d53a2de9e7c156fad6f9857e3d59318d64513929 send MYTH to Mythos for example.

IIRC the asset_hub_fee we set is 0.1 DOT.

Events on AH in subscan from https://assethub-polkadot.subscan.io/block/7056042?event=7056042-7&tab=event we can see `[polkadotxcm (AssetsTrapped)]` event with 0.05 DOT trapped which is more than half of the fee. 

The transport fees will be sent to [Treasury](https://assethub-polkadot.subscan.io/account/14xmwinmCEz6oRrFdczHKqHgWNMiCysE2KrA4jXXAAM1Eogk) corresponds to `[balances (Minted)]` event with 0.03 DOT.

Which means the transport fee is even more than the xcm execution which is 0.1-0.05-0.03=0.02 DOT.

So 1/2 is safer than 3/4 IMO but anyway the magic number makes things difficult to understand. With Adrian's https://github.com/paritytech/polkadot-sdk/pull/5563 I still prefer to use `SetFeesMode`.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5546/commits/991de7c153bb2792b42ab1f7a825658aaf5801c3

Reply:
Yeah I am fine with this but please give more details about the problem in the comment above `SetFeesMode` instruction.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5546/commits/d5b9b226477dcce85261f34e5bf6bfa67913619a